### PR TITLE
kasan: use init value to avoid kasan used before bss clear

### PR DESCRIPTION
**Pull request by GUIDINGLI**
## Summary

kasan: use init value to avoid kasan used before bss clear

## Impact

kasan

## Testing

VELA

### CODE REVIEW

File: mm/kasan/kasan.c [hunk 1]
```diff
@@ -79,6 +82,11 @@ static FAR uintptr_t *kasan_mem_to_shadow(FAR const void *ptr, size_t size,
   FAR struct kasan_region_s *region;
   uintptr_t addr = (uintptr_t)ptr;
 
+  if (g_region_init != KASAN_INIT_VALUE)
```
Comment (pkarashchenko) [inline mm/kasan/kasan.c:85]: Why not just use `bool`?

Comment (xiaoxiang781216) [reply mm/kasan/kasan.c:85]: The code instrumented by compiler's KASAN may run before the startup initialize .bss or .data section. Using the bigger number increase the possiblity to catch the premature access before KASAN subsystem finish the initialization.

**Review by xiaoxiang781216 — APPROVED** [review]

File: mm/kasan/kasan.c (hunk 1 above)
Comment (pkarashchenko) [reply mm/kasan/kasan.c:85]: So basically `g_region_init` can be a random value. But what will happen if the device is reseted, but memory content retains?

Comment (xiaoxiang781216) [reply mm/kasan/kasan.c:85]: it depends on the hardware design, so the approach just adds an additional safeguard.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**